app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1036,11 +1036,26 @@ class PatientRepositoryAndroidTest {
     assertEquals(creationTimeOf2ndBp, recentPatientAfter2ndBp.lastBp!!.createdAt)
   }
 
+  @Test
+  fun verify_deleted_bps_are_not_picked_up() {
+    val facilityUuid = testData.qaUserFacilityUuid()
+    savePatientWithBp(facilityUuid = facilityUuid)
+    savePatientWithBp(facilityUuid = facilityUuid, deletedAt = Instant.now())
+    savePatientWithBp(facilityUuid = facilityUuid)
+
+    val recentPatientsCount = patientRepository
+        .recentPatients(facilityUuid, limit = 10)
+        .blockingFirst()
+        .size
+    assertEquals(2, recentPatientsCount)
```

Comment:
Can you water-tight this test further to verify that the 2 records are indeed the expected ones?